include/cvc5/cvc5_proof_rule.h

```diff
@@ -2481,6 +2481,93 @@ enum ENUM(ProofRewriteRule) : uint32_t
    * \endverbatim
    */
   EVALUE(DT_CONS_EQ),
+
+  /**
+   * \verbatim embed:rst:leading-asterisk
+   * **Bitvectors - Unsigned multiplication overflow detection elimination**
+   *
+   * See M.Gok, M.J. Schulte, P.I. Balzola, "Efficient integer multiplication
+   * overflow detection circuits", 2001.
+   * http://ieeexplore.ieee.org/document/987767
+   * \endverbatim
+   */
+  EVALUE(BV_UMULO_ELIMINATE),
+  /**
+   * \verbatim embed:rst:leading-asterisk
+   * **Bitvectors - Signed multiplication overflow detection elimination**
+   *
+   * See M.Gok, M.J. Schulte, P.I. Balzola, "Efficient integer multiplication
+   * overflow detection circuits", 2001.
+   * http://ieeexplore.ieee.org/document/987767
+   * \endverbatim
+   */
+  EVALUE(BV_SMULO_ELIMINATE),
+  /**
+   * \verbatim embed:rst:leading-asterisk
+   * **Bitvectors - Flatten nested add/or/mult/xor/and expression**
+   *
+   * .. math::
+   *    ((a1 and a2) and (a3 and a4)) \to (a1 and a2 and a3 and a4)
+   *
+   * \endverbatim
+   */
+  EVALUE(BV_FLATTEN_ASSOC_COMMUT),
+  /**
+   * \verbatim embed:rst:leading-asterisk
+   * **Bitvectors - Flatten nested add/or/mult/xor/and expression**
+   *
+   * .. math::
+   *    ((a1 and a2) and (a3 and a4)) \to (a1 and a2 and a3 and a4)
+   *
+   * \endverbatim
+   */
+  EVALUE(BV_FLATTEN_ASSOC_COMMUT_NO_DUPLICATES),
+  /**
+   * \verbatim embed:rst:leading-asterisk
+   * **Bitvectors - Combine like terms during addition by counting terms**
+   * \endverbatim
+   */
+  EVALUE(BV_ADD_COMBINE_LIKE_TERMS),
+  /**
+   * \verbatim embed:rst:leading-asterisk
+   * **Bitvectors - Extract negations from multiplicands**
+   *
+   * .. math::
+   *    (-a bvmul b bvmul c) \to -(a bvmul b c)
+   *
+   * \endverbatim
+   */
+  EVALUE(BV_MULT_SIMPLIFY),
+  /**
+   * \verbatim embed:rst:leading-asterisk
+   * **Bitvectors - Remove like terms on both sides of an equation**
+   *
+   * .. math::
+   *    a + \dots = a + \dots \to \dots = \dots
+   *
+   * \endverbatim
+   */
+  EVALUE(BV_SOLVE_EQ),
```

Comment:
Are we doing some kind of normalization here?

Reply:
I don't know I just copied these from the proof rule code